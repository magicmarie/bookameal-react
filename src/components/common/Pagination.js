import React, { Component } from "react";
import PropTypes from "prop-types";

//Pagination component to be reused in other components where pagination is needed.

class Pagination extends Component {
  static propTypes = {
    pages: PropTypes.number,
    currentPage: PropTypes.number,
    nextPage: PropTypes.number,
    previousPage: PropTypes.number,
    changePage: PropTypes.func.isRequired
  };

  static defaultProps = {
    nextPage: null,
    previousPage: null,
    totalPages: null,
    currentPage: null
  };

  changePage = event => {
    // preventing default link behaviour
    event.preventDefault();
    // do nothing if the selected page doesnot exist and is not a number
    if (Number.isNaN(Number(event.target.dataset.page))) {
      return;
    }
    // call the change page function if selected page exists
    this.props.changePage(Number(event.target.dataset.page));
  };

  render() {
    const { pages, currentPage, nextPage, previousPage } = this.props;

    //   generating the pages
    const totalpages = Array.from({ length: pages }).map((item, index) => (
      <li
        className={`page-item${index + 1 === currentPage ? " active" : ""}`}
        key={index}
      >
        <a
          className="page-link"
          href="!#"
          data-page={index + 1}
          onClick={this.changePage}
        >
          {index + 1}
        </a>
      </li>
    ));

    return (
      <nav aria-label="Page navigation example justify-content">
        <ul className="pagination">
          <li className={`page-item${previousPage ? "" : " disabled"}`}>
            <a
              className="page-link"
              data-page={previousPage}
              onClick={this.changePage}
              href="!#"
            >
              Previous
            </a>
          </li>
          {totalpages}
          <li className={`page-item${nextPage ? "" : " disabled"}`}>
            <a
              className="page-link mb-4"
              href="!#"
              data-page={nextPage}
              onClick={this.changePage}
            >
              Next
            </a>
          </li>
        </ul>
      </nav>
    );
  }
}

export default Pagination;
