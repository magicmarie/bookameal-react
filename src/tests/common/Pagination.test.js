import React from "react";
import { shallow } from "enzyme";
import Pagination from "../../components/common/Pagination";

describe("Pagination component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Pagination pages={10} currentPage={9} nextPage={10} previousPage={8} changePage={jest.fn()} event={jest.fn()} />);
  });
  it("renders successfully", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders successfully", () => {
    const handleEvent = {
      preventDefault: jest.fn(),
      target: { dataset: "x" }
    };
    wrapper.find(".mb-4").simulate("click", handleEvent);
  });
  it("renders successfully", () => {
    const handleEvent = {
      preventDefault: jest.fn(),
      target: { dataset: 8 }
    };
    wrapper.find("#previous").simulate("click", handleEvent);
  });
});
