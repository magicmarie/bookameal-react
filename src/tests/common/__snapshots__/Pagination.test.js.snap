// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component renders successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    changePage={[Function]}
    currentPage={9}
    event={[Function]}
    nextPage={10}
    pages={10}
    previousPage={8}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "Page navigation example",
      "children": <ul
        className="pagination"
>
        <li
                className="page-item"
        >
                <a
                        className="page-link"
                        data-page={8}
                        href="!#"
                        id="previous"
                        onClick={[Function]}
                >
                        Previous
                </a>
        </li>
        <li
                className="page-item"
        >
                <a
                        className="page-link"
                        data-page={1}
                        href="!#"
                        onClick={[Function]}
                >
                        1
                </a>
        </li>
        <li
                className="page-item"
        >
                <a
                        className="page-link"
                        data-page={2}
                        href="!#"
                        onClick={[Function]}
                >
                        2
                </a>
        </li>
        <li
                className="page-item"
        >
                <a
                        className="page-link"
                        data-page={3}
                        href="!#"
                        onClick={[Function]}
                >
                        3
                </a>
        </li>
        <li
                className="page-item"
        >
                <a
                        className="page-link"
                        data-page={4}
                        href="!#"
                        onClick={[Function]}
                >
                        4
                </a>
        </li>
        <li
                className="page-item"
        >
                <a
                        className="page-link"
                        data-page={5}
                        href="!#"
                        onClick={[Function]}
                >
                        5
                </a>
        </li>
        <li
                className="page-item"
        >
                <a
                        className="page-link"
                        data-page={6}
                        href="!#"
                        onClick={[Function]}
                >
                        6
                </a>
        </li>
        <li
                className="page-item"
        >
                <a
                        className="page-link"
                        data-page={7}
                        href="!#"
                        onClick={[Function]}
                >
                        7
                </a>
        </li>
        <li
                className="page-item"
        >
                <a
                        className="page-link"
                        data-page={8}
                        href="!#"
                        onClick={[Function]}
                >
                        8
                </a>
        </li>
        <li
                className="page-item active"
        >
                <a
                        className="page-link"
                        data-page={9}
                        href="!#"
                        onClick={[Function]}
                >
                        9
                </a>
        </li>
        <li
                className="page-item"
        >
                <a
                        className="page-link"
                        data-page={10}
                        href="!#"
                        onClick={[Function]}
                >
                        10
                </a>
        </li>
        <li
                className="page-item"
        >
                <a
                        className="page-link mb-4"
                        data-page={10}
                        href="!#"
                        onClick={[Function]}
                >
                        Next
                </a>
        </li>
</ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="page-item"
>
            <a
                        className="page-link"
                        data-page={8}
                        href="!#"
                        id="previous"
                        onClick={[Function]}
            >
                        Previous
            </a>
</li>,
          Array [
            <li
              className="page-item"
>
              <a
                            className="page-link"
                            data-page={1}
                            href="!#"
                            onClick={[Function]}
              >
                            1
              </a>
</li>,
            <li
              className="page-item"
>
              <a
                            className="page-link"
                            data-page={2}
                            href="!#"
                            onClick={[Function]}
              >
                            2
              </a>
</li>,
            <li
              className="page-item"
>
              <a
                            className="page-link"
                            data-page={3}
                            href="!#"
                            onClick={[Function]}
              >
                            3
              </a>
</li>,
            <li
              className="page-item"
>
              <a
                            className="page-link"
                            data-page={4}
                            href="!#"
                            onClick={[Function]}
              >
                            4
              </a>
</li>,
            <li
              className="page-item"
>
              <a
                            className="page-link"
                            data-page={5}
                            href="!#"
                            onClick={[Function]}
              >
                            5
              </a>
</li>,
            <li
              className="page-item"
>
              <a
                            className="page-link"
                            data-page={6}
                            href="!#"
                            onClick={[Function]}
              >
                            6
              </a>
</li>,
            <li
              className="page-item"
>
              <a
                            className="page-link"
                            data-page={7}
                            href="!#"
                            onClick={[Function]}
              >
                            7
              </a>
</li>,
            <li
              className="page-item"
>
              <a
                            className="page-link"
                            data-page={8}
                            href="!#"
                            onClick={[Function]}
              >
                            8
              </a>
</li>,
            <li
              className="page-item active"
>
              <a
                            className="page-link"
                            data-page={9}
                            href="!#"
                            onClick={[Function]}
              >
                            9
              </a>
</li>,
            <li
              className="page-item"
>
              <a
                            className="page-link"
                            data-page={10}
                            href="!#"
                            onClick={[Function]}
              >
                            10
              </a>
</li>,
          ],
          <li
            className="page-item"
>
            <a
                        className="page-link mb-4"
                        data-page={10}
                        href="!#"
                        onClick={[Function]}
            >
                        Next
            </a>
</li>,
        ],
        "className": "pagination",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link"
              data-page={8}
              href="!#"
              id="previous"
              onClick={[Function]}
>
              Previous
</a>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Previous",
              "className": "page-link",
              "data-page": 8,
              "href": "!#",
              "id": "previous",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Previous",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link"
              data-page={1}
              href="!#"
              onClick={[Function]}
>
              1
</a>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
              "className": "page-link",
              "data-page": 1,
              "href": "!#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 1,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link"
              data-page={2}
              href="!#"
              onClick={[Function]}
>
              2
</a>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "className": "page-link",
              "data-page": 2,
              "href": "!#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 2,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link"
              data-page={3}
              href="!#"
              onClick={[Function]}
>
              3
</a>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 3,
              "className": "page-link",
              "data-page": 3,
              "href": "!#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 3,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link"
              data-page={4}
              href="!#"
              onClick={[Function]}
>
              4
</a>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "className": "page-link",
              "data-page": 4,
              "href": "!#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 4,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link"
              data-page={5}
              href="!#"
              onClick={[Function]}
>
              5
</a>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 5,
              "className": "page-link",
              "data-page": 5,
              "href": "!#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 5,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link"
              data-page={6}
              href="!#"
              onClick={[Function]}
>
              6
</a>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 6,
              "className": "page-link",
              "data-page": 6,
              "href": "!#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 6,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link"
              data-page={7}
              href="!#"
              onClick={[Function]}
>
              7
</a>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 7,
              "className": "page-link",
              "data-page": 7,
              "href": "!#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 7,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link"
              data-page={8}
              href="!#"
              onClick={[Function]}
>
              8
</a>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 8,
              "className": "page-link",
              "data-page": 8,
              "href": "!#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 8,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link"
              data-page={9}
              href="!#"
              onClick={[Function]}
>
              9
</a>,
            "className": "page-item active",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 9,
              "className": "page-link",
              "data-page": 9,
              "href": "!#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 9,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link"
              data-page={10}
              href="!#"
              onClick={[Function]}
>
              10
</a>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 10,
              "className": "page-link",
              "data-page": 10,
              "href": "!#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 10,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="page-link mb-4"
              data-page={10}
              href="!#"
              onClick={[Function]}
>
              Next
</a>,
            "className": "page-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next",
              "className": "page-link mb-4",
              "data-page": 10,
              "href": "!#",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Next",
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Page navigation example",
        "children": <ul
          className="pagination"
>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              data-page={8}
                              href="!#"
                              id="previous"
                              onClick={[Function]}
                    >
                              Previous
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              data-page={1}
                              href="!#"
                              onClick={[Function]}
                    >
                              1
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              data-page={2}
                              href="!#"
                              onClick={[Function]}
                    >
                              2
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              data-page={3}
                              href="!#"
                              onClick={[Function]}
                    >
                              3
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              data-page={4}
                              href="!#"
                              onClick={[Function]}
                    >
                              4
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              data-page={5}
                              href="!#"
                              onClick={[Function]}
                    >
                              5
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              data-page={6}
                              href="!#"
                              onClick={[Function]}
                    >
                              6
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              data-page={7}
                              href="!#"
                              onClick={[Function]}
                    >
                              7
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              data-page={8}
                              href="!#"
                              onClick={[Function]}
                    >
                              8
                    </a>
          </li>
          <li
                    className="page-item active"
          >
                    <a
                              className="page-link"
                              data-page={9}
                              href="!#"
                              onClick={[Function]}
                    >
                              9
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              data-page={10}
                              href="!#"
                              onClick={[Function]}
                    >
                              10
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link mb-4"
                              data-page={10}
                              href="!#"
                              onClick={[Function]}
                    >
                              Next
                    </a>
          </li>
</ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="page-item"
>
              <a
                            className="page-link"
                            data-page={8}
                            href="!#"
                            id="previous"
                            onClick={[Function]}
              >
                            Previous
              </a>
</li>,
            Array [
              <li
                className="page-item"
>
                <a
                                className="page-link"
                                data-page={1}
                                href="!#"
                                onClick={[Function]}
                >
                                1
                </a>
</li>,
              <li
                className="page-item"
>
                <a
                                className="page-link"
                                data-page={2}
                                href="!#"
                                onClick={[Function]}
                >
                                2
                </a>
</li>,
              <li
                className="page-item"
>
                <a
                                className="page-link"
                                data-page={3}
                                href="!#"
                                onClick={[Function]}
                >
                                3
                </a>
</li>,
              <li
                className="page-item"
>
                <a
                                className="page-link"
                                data-page={4}
                                href="!#"
                                onClick={[Function]}
                >
                                4
                </a>
</li>,
              <li
                className="page-item"
>
                <a
                                className="page-link"
                                data-page={5}
                                href="!#"
                                onClick={[Function]}
                >
                                5
                </a>
</li>,
              <li
                className="page-item"
>
                <a
                                className="page-link"
                                data-page={6}
                                href="!#"
                                onClick={[Function]}
                >
                                6
                </a>
</li>,
              <li
                className="page-item"
>
                <a
                                className="page-link"
                                data-page={7}
                                href="!#"
                                onClick={[Function]}
                >
                                7
                </a>
</li>,
              <li
                className="page-item"
>
                <a
                                className="page-link"
                                data-page={8}
                                href="!#"
                                onClick={[Function]}
                >
                                8
                </a>
</li>,
              <li
                className="page-item active"
>
                <a
                                className="page-link"
                                data-page={9}
                                href="!#"
                                onClick={[Function]}
                >
                                9
                </a>
</li>,
              <li
                className="page-item"
>
                <a
                                className="page-link"
                                data-page={10}
                                href="!#"
                                onClick={[Function]}
                >
                                10
                </a>
</li>,
            ],
            <li
              className="page-item"
>
              <a
                            className="page-link mb-4"
                            data-page={10}
                            href="!#"
                            onClick={[Function]}
              >
                            Next
              </a>
</li>,
          ],
          "className": "pagination",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link"
                data-page={8}
                href="!#"
                id="previous"
                onClick={[Function]}
>
                Previous
</a>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Previous",
                "className": "page-link",
                "data-page": 8,
                "href": "!#",
                "id": "previous",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Previous",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link"
                data-page={1}
                href="!#"
                onClick={[Function]}
>
                1
</a>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "className": "page-link",
                "data-page": 1,
                "href": "!#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 1,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link"
                data-page={2}
                href="!#"
                onClick={[Function]}
>
                2
</a>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2,
                "className": "page-link",
                "data-page": 2,
                "href": "!#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 2,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link"
                data-page={3}
                href="!#"
                onClick={[Function]}
>
                3
</a>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 3,
                "className": "page-link",
                "data-page": 3,
                "href": "!#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 3,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link"
                data-page={4}
                href="!#"
                onClick={[Function]}
>
                4
</a>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 4,
                "className": "page-link",
                "data-page": 4,
                "href": "!#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 4,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link"
                data-page={5}
                href="!#"
                onClick={[Function]}
>
                5
</a>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 5,
                "className": "page-link",
                "data-page": 5,
                "href": "!#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 5,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link"
                data-page={6}
                href="!#"
                onClick={[Function]}
>
                6
</a>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 6,
                "className": "page-link",
                "data-page": 6,
                "href": "!#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 6,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link"
                data-page={7}
                href="!#"
                onClick={[Function]}
>
                7
</a>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 7,
                "className": "page-link",
                "data-page": 7,
                "href": "!#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 7,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "7",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link"
                data-page={8}
                href="!#"
                onClick={[Function]}
>
                8
</a>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 8,
                "className": "page-link",
                "data-page": 8,
                "href": "!#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 8,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "8",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link"
                data-page={9}
                href="!#"
                onClick={[Function]}
>
                9
</a>,
              "className": "page-item active",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 9,
                "className": "page-link",
                "data-page": 9,
                "href": "!#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 9,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "9",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link"
                data-page={10}
                href="!#"
                onClick={[Function]}
>
                10
</a>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 10,
                "className": "page-link",
                "data-page": 10,
                "href": "!#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 10,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="page-link mb-4"
                data-page={10}
                href="!#"
                onClick={[Function]}
>
                Next
</a>,
              "className": "page-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Next",
                "className": "page-link mb-4",
                "data-page": 10,
                "href": "!#",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Next",
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
